### ISSUE TITLE
YOLOv11 OBB model cannot learn large size labeled OBB

### ISSUE THREAD

**Issue by LeMinhNgan** [ultralytics/ultralytics#22588]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I am training YOLOv11 OBB model with pretrained yolo11x-obb. The labeled obb size is about 80% of image size (3000x4000). I set imgsz=2880 to get the best accuracy. However the MAP50-95 never hit over 0.15
I see predicted images and found the predicted box size is quite small and not fit the object. I guess there is a limitation of the bounding box size, but don't know how to fix it. Can anyone help me to solve it?

<img width="1420" height="591" alt="Image" src="https://github.com/user-attachments/assets/91b269da-6a0b-4fb9-b367-7190315fc87d" />

### Environment

ultralytics 8.3.222
python 3.10
Ultra 5, RAM 192GB, RTX 5090, Windows11

### Minimal Reproducible Example

n/a

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @LeMinhNgan, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get moving quickly—an Ultralytics engineer will follow up soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Since you’re training a YOLO11 OBB model and seeing unexpectedly small predictions, please include:
-Exact training/validation commands used (with all args like `imgsz`, `device`, `workers`, `batch`, `patience`, `augment`, `rect`, `conf`, etc.)
-Exact model/checkpoint name (e.g., `yolo11x-obb.pt`) and the `ultralytics` version you ran
-A tiny dataset slice (2–3 images) with corresponding OBB labels to reproduce
-Screenshots or files: sample train/val predictions and training logs (metrics per epoch)
-Your YAMLs (`data.yaml`, any custom `hyp.yaml`) and note if mosaic/flip/scale are enabled
-Confirm if results persist with default settings and a smaller `imgsz` baseline

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). 🙌

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit. ✅

Thanks again for the detailed report and the screenshot. Once you share the MRE and training details, we’ll help investigate further. An Ultralytics engineer will assist you here shortly. 🛠️

**Comment by LeMinhNgan** [comment]
**Training detail as following:**

```
from ultralytics import YOLO

data= r'D:\py_files\obb_yolov11.yaml'      
epochs= 1000
imgsz= 2880                                
batch= 1                                    
prerained= r'D:\py_files\yolo11x-obb.pt'    
     
if __name__ == "__main__":
    model= YOLO(model= prerained)
    results= model.train(data= data, epochs= epochs, imgsz= imgsz, batch= batch) 
```

**The YAML file:**
```
path: C:/Users/ngan.le/OneDrive - Coherent Corporation/Documents/OBB dataset
train: train/images
val: val/images
nc: 1
names: ['tray']
```

**Dataset:** 432 images with labels, image size is about 3000x4000

**Comment by LeMinhNgan** [comment]
> If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit. ✅
> 
> Thanks again for the detailed report and the screenshot. Once you share the MRE and training details, we’ll help investigate further. An Ultralytics engineer will assist you here shortly. 🛠️

I give more detail of training above. The training stop after apprx 300 epochs due to no improvement.

**Comment by Y-T-G** [comment]
If you have large objects, you should be using smaller `imgsz`. Otherwise the receptive field can't see the whole object. I don't understand the need for `imgsz=2880` when your object is already big. It's not going to improve accuracy. Large `imgsz` mostly improves accuracy of small objects, not large objects.

**Comment by AI-Passionner** [comment]
@LeMinhNgan you can have a try of YOLOV8 in the latest ultralytics or even older, ultralytics==8.2.92. or give the training a large patience= 300 or no early stop

**Comment by LeMinhNgan** [comment]
> If you have large objects, you should be using smaller `imgsz`. Otherwise the receptive field can't see the whole object. I don't understand the need for `imgsz=2880` when your object is already big. It's not going to improve accuracy. Large `imgsz` mostly improves accuracy of small objects, not large objects.

@Y-T-G I will use this model for an Automation project of Pick-n-Place. The acceptable tolerance is 2 pixels only. I'm afraid that the prediction will not be pixel-wise precise if "imgsz" is small

What is the largest of "imgsz" so that the receptive field can see the whole object? could you please advise the number?

**Comment by Y-T-G** [comment]
You will need to use P6 layer and also increase `reg_max`. 

https://github.com/ultralytics/ultralytics/blob/ac9af2e437bf5370b32191438bb56c632da93ec4/ultralytics/nn/modules/head.py#L89
https://github.com/Y-T-G/community/blob/main/cfg/obb/yolo11-p6.yaml
But even then, this size would probably be too large.

**Comment by LeMinhNgan** [comment]
> Line 89 in [ac9af2e](/ultralytics/ultralytics/commit/ac9af2e437bf5370b32191438bb56c632da93ec4)

@Y-T-G  Many thanks!
